packages/plugin-typeorm/src/typeorm.ts

```diff
@@ -0,0 +1,149 @@
+import * as typeorm from 'typeorm';
+
+const VERSION = '0.0.1';
+
+class TypeormPlugin extends BasePlugin<typeof typeorm> {
+    protected _config!: TypeormPluginConfig;
+
+    constructor(readonly moduleName: string) {
+        super(`opentelemetry-plugin-typeorm`, VERSION);
+    }
+
+    protected patch(): typeof typeorm {
+        this._logger.debug(`applying patch to ${this.moduleName}@${this.version}`);
+        shimmer.wrap(
+            this._moduleExports.ConnectionManager.prototype,
+            'create',
+            this._createConnectionManagerPatch.bind(this)
+        );
+
+        return this._moduleExports;
+    }
+    protected unpatch(): void {
+        shimmer.unwrap(this._moduleExports.ConnectionManager.prototype, 'create');
+    }
+
+    private _createConnectionManagerPatch(original: (options: typeorm.ConnectionOptions) => typeorm.Connection) {
+        const thisPlugin = this;
+        return function (options: typeorm.ConnectionOptions) {
+            const connection: typeorm.Connection = original.apply(this, arguments);
+
+            // Both types using same patch right now, keep different declarations for future improvements
+            const functionsUsingEntityPersistExecutor = ['save', 'remove', 'softRemove', 'recover'];
+            const functionsUsingQueryBuilder = [
+                'insert',
+                'update',
+                'delete',
+                'softDelete',
+                'restore',
+                'count',
+                'find',
+                'findAndCount',
+                'findByIds',
+                'findOne',
+                'increment',
+                'decrement',
+            ];
+
+            functionsUsingEntityPersistExecutor.forEach((op) =>
+                shimmer.wrap(
+                    connection.manager,
+                    op as keyof typeorm.EntityManager,
+                    thisPlugin._getEntityManagerFunctionPatch(op).bind(thisPlugin)
+                )
+            );
+
+            functionsUsingQueryBuilder.forEach((op) =>
+                shimmer.wrap(
+                    connection.manager,
+                    op as keyof typeorm.EntityManager,
+                    thisPlugin._getEntityManagerFunctionPatch(op).bind(thisPlugin)
+                )
+            );
+
+            return connection;
+        };
+    }
+
+    private _getEntityManagerFunctionPatch(opName: string) {
+        const thisPlugin = this;
+        thisPlugin._logger.debug(`TypeormPlugin: patched EntityManager ${opName} prototype`);
+        return function (original: Function) {
+            return async function (...args: any[]) {
+                const connectionOptions = this?.connection?.options ?? {}
+                const attributes = {
+                    [DatabaseAttribute.DB_SYSTEM]: connectionOptions.type,
+                    [DatabaseAttribute.DB_USER]: connectionOptions.username,
+                    // [GeneralAttribute.NET_PEER_IP]: '?',
```

Comment:
Do we need those comments?

Reply:
yap Amir is right

Reply:
Sounds good, consider to add a comment to explain it.